backend/src/app.ts

```diff
@@ -51,12 +59,16 @@ if (process.env.COMPILER == CompilerTypes.JDOODLE) {
 // initialize database connection
 try {
     const client = new MongoClient(process.env.DATABASE_URI as string);
-
     await client.connect();
-
     const database = client.db(process.env.DATABASE_NAME);
+
     projectCollection = database.collection<Project>("projects");
     converterCollection = database.collection<Converter>("converters");
+
+    // unique constraint on provider and user id from provider
+    userCollection = database.collection<User>("users");
+    await userCollection.createIndex({ provider: 1, id: 1 }, { unique: true });
```

Comment:
Hmm czy to nie wystarczy raz utworzyć taki index? To się wykonuje za każdym razem jak uruchamiamy backend, może to lepiej raz ręcznie ustawić zamiast trzymać w kodzie?

Reply:
I ten index to powinnien być po `_id` zamiast `id`, w przeciwnym wypadku, przy próbie utworzenia drugiego konta dostawałem taki error:
`MongoServerError: E11000 duplicate key error collection: AlgoDebug_Marcin_Wojdat.users index: provider_1_id_1 dup key: { provider: "google", id: null }`